Make escape cancel edition and restore pristine value

oliverguenther opened the pull request:
This lets escape while in edit mode cancel the edition and restore the value using a pristine copy of the value.

I was trying to use the pristine values from `form.payload`, however they aren't `$load`ed , and thus are not complete values. I argue its easier to copy the value when starting to edit the field.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-feature-escape-cancels.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-feature-escape-cancels.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther commented:
@ulferts Merge this for many wins.

oliverguenther commented:
@furinvader Can you take a look at this PR?

furinvader commented:
alles falsch
fang von forne ann

oliverguenther committed:
Skip activation when already activated

ng-aria probably adds keypress handlers to ng-click handlers for
screenreaders, such as for space. This triggers `activate` of the inline
edit and causes the pristine copy to be invalid.

oliverguenther committed:
Make escape cancel edition and restore pristine value

oliverguenther commented:
Merging this when travis agrees after fixing the bug from Alex' review.
